db/local_db/status.go

```diff
@@ -36,3 +42,41 @@ func GetStatus() (*RunningDBInstanceInfo, error) {
 
 	return info, nil
 }
+
+// errorIfUnknownService errors if it can find a `postmaster.pid` in the `INSTALL_DIR`
+// and the PID recorded in the found `postmaster.pid` is running
+func errorIfUnknownService() error {
+	// no postmaster.pid, we are good
+	if !helpers.FileExists(getPostmasterPidLocation()) {
+		return nil
+	}
+
+	// read the content of the postmaster.pid file
+	fileContent, err := ioutil.ReadFile(getPostmasterPidLocation())
+	if err != nil {
+		return err
+	}
+
+	// the first line contains the PID
+	lines := strings.FieldsFunc(string(fileContent), func(r rune) bool {
+		return r == '\n'
```

Comment:
is this valid cross platform?

Reply:
as per https://github.com/postgres/postgres/blob/76e38b37a5f179d4c9d2865ff31b79130407530b/src/bin/pg_ctl/pg_ctl.c#L317

`pg_ctl` uses `\n` to split lines in the `postmaster.pid` file